tests/python_package_test/test_engine.py

```diff
@@ -1432,7 +1434,7 @@ def test_feature_name():
     assert feature_names == gbm.feature_name()
 
 
-def test_feature_name_with_non_ascii(rng):
+def test_feature_name_with_non_ascii(rng, model_path_txt):
```

Comment:
As I mentioned above... let's please just use `tmp_path` here.

Assign a variable with the model file path and re-use it in both `gbm.save_model()` and `lgb.Booster()` below, something like this:

```python
model_path_txt = tmp_path / "lgb.model"
gbm.save_model(model_path_txt)

gbm2 = lgb.Booster(model_file=model_path_txt)
```